test/h-group/level-11.js

```diff
@@ -0,0 +1,345 @@
+		// Alice must have r3, r2
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.ALICE][0].order], ['r3']);
+		ExAsserts.cardHasInferences(game.common.thoughts[game.state.hands[PLAYER.ALICE][1].order], ['r2']);
+	});
+
+	it(`doesn't bluff through self finesses`, () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['y1', 'b5', 'y1', 'r4', 'y4'],
+			['p2', 'p3', 'g3', 'g4', 'y5'],
+		], {
+			level: 11
+		});
+
+		const { play_clues } = find_clues(game);
+		const bluff_clues = play_clues[2].filter(clue => {
+			return clue.type == CLUE.RANK && clue.target == 2 && clue.value == 3;
+		});
+		assert.equal(bluff_clues.length, 0);
+	});
+
+	it(`doesn't bluff when bluff can't be known by next player to play`, () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['y1', 'b5', 'y1', 'r4', 'y4'],
+			['p2', 'r2', 'b2', 'g4', 'y5'],
+		], {
+			level: 11
+		});
+
+		const { play_clues } = find_clues(game);
+		const bluff_clues = play_clues[2].filter(clue => {
+			return clue.type == CLUE.RANK && clue.target == 2 && clue.value == 2;
+		});
+		assert.equal(bluff_clues.length, 0);
+	});
+
+	it(`doesn't bluff on top of unknown queued cards`, () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['g1', 'r2', 'y1', 'y1', 'r4'],
+			['p2', 'b2', 'r3', 'y5', 'y4'],
+			['g1', 'g2', 'g3', 'g5', 'p4'],
+		], {
+			level: 11,
+			play_stacks: [1, 0, 0, 0, 0],
+			starting: PLAYER.DONALD
+		});
+		takeTurn(game, 'Donald clues red to Cathy');
+		// With g1, r2 already queued, we cannot bluff the y1.
+		const { play_clues } = find_clues(game);
+		const bluff_clues = play_clues[2].filter(clue => {
+			return clue.type == CLUE.RANK && clue.target == 2 && clue.value == 2 ||
+				clue.type == CLUE.COLOUR && clue.target == 2 && (clue.value == 3 || clue.value == 4);
+		});
+		assert.equal(bluff_clues.length, 0);
+	});
+
+	it(`understands a bluff on top of known queued plays`, () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['r1', 'r2', 'y1', 'y1', 'r4'],
+			['g5', 'b2', 'r3', 'y5', 'y4'],
+			['g1', 'g2', 'g3', 'g5', 'p4'],
+		], {
+			level: 11,
+			play_stacks: [0, 0, 0, 0, 0],
+			starting: PLAYER.DONALD
+		});
+		takeTurn(game, 'Donald clues red to Bob');
+		// Since Bob has known plays a bluff should be possible on blue.
+		takeTurn(game, 'Alice clues blue to Cathy');
+
+		assert.equal(game.common.thoughts[game.state.hands[PLAYER.BOB][2].order].finessed, true);
+	});
+
+	it(`doesn't bluff on top of known cards which might match bluff`, () => {
+		const game = setup(HGroup, [
+			['xx', 'xx', 'xx', 'xx', 'xx'],
+			['r1', 'g1', 'y1', 'y1', 'y5'],
+			['p4', 'b2', 'r3', 'y5', 'y4'],
+			['g1', 'g2', 'g3', 'g5', 'p4'],
+		], {
+			level: 11,
+			play_stacks: [0, 0, 0, 0, 0],
+			starting: PLAYER.DONALD
+		});
+		takeTurn(game, 'Donald clues red to Bob');
+		// r2 is known to be queued, and would be played over finesse slot.
```

Comment:
It doesn't look like this test matches the name, assuming I understand your intent. Only r1 is queued in Bob's hand, and the reason why 2 to Cathy is not permitted is because g1 would look like it connects to b2, not because r1 would be played instead.

Maybe these are closer to what you were going for? The code passes both of them.
```js
it(`doesn't bluff on top of colour-clued cards which might match bluff`, () => {
	const game = setup(HGroup, [
		['xx', 'xx', 'xx', 'xx', 'xx'],
		['r1', 'r4', 'y1', 'y1', 'y5'],
		['p4', 'b2', 'r3', 'y5', 'y4'],
	], { level: 11 });

	takeTurn(game, 'Alice clues red to Bob');
	takeTurn(game, 'Bob plays r1', 'g1');
	takeTurn(game, 'Cathy clues 5 to Bob');

	// Alice cannot use r3 to bluff Bob's g1, as r4 would play instead.
	const { play_clues } = find_clues(game);
	assert.ok(!play_clues[PLAYER.CATHY].some(clue =>
		(clue.type === CLUE.RANK && clue.value === 3) ||
		(clue.type === CLUE.COLOUR && clue.value === COLOUR.RED)));
});

it(`doesn't bluff on top of rank-clued cards which might match bluff`, () => {
	const game = setup(HGroup, [
		['xx', 'xx', 'xx', 'xx', 'xx'],
		['y2', 'r2', 'y1', 'y1', 'y5'],
		['g4', 'b2', 'p3', 'y5', 'y4'],
	], {
		level: 11,
		play_stacks: [0, 1, 0, 0, 1]
	});

	takeTurn(game, 'Alice clues 2 to Bob');
	takeTurn(game, 'Bob plays y2', 'g1');
	takeTurn(game, 'Cathy clues 5 to Bob');

	// Alice cannot use p3 to bluff Bob's g1, as r2 would play instead.
	const { play_clues } = find_clues(game);
	assert.ok(!play_clues[PLAYER.CATHY].some(clue =>
		(clue.type === CLUE.RANK && clue.value === 3) ||
		(clue.type === CLUE.COLOUR && clue.value === COLOUR.PURPLE)));
});
```

Reply:
Thanks, yeah this is exactly what I was thinking, added!